Don't specify source and target in parent POM for maven-compiler-plugin

Tycho will take it from the project preferences and we avoid problems in #222 (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=567305)

It hasn't been a good idea: Sonar scanner does not know about the Java version and we have a lot of parsing errors in tests, which will then not be used for unit test reports, e.g.,

```
[ERROR] Unable to parse source file : 'edelta/src/edelta/interpreter/EdeltaInterpreterHelper.java'
[ERROR] Parse error at line 57 column 11: Multi-catch parameters are not allowed for source level below 1.7
[ERROR] Unable to parse source file : 'edelta/src/edelta/compiler/EdeltaCompilerUtil.java'
[ERROR] Parse error at line 44 column 3: Lambda expressions are allowed only at source level 1.8 or above
[ERROR] Unable to parse source file : 'edelta/src/edelta/interpreter/EdeltaInterpreterResourceListener.java'
[ERROR] Parse error at line 134 column 18: Method references are allowed only at source level 1.8 or above
[ERROR] Unable to parse source file : 'edelta/src/edelta/resource/derivedstate/EdeltaDerivedStateHelper.java'
[ERROR] Parse error at line 89 column 54: Lambda expressions are allowed only at source level 1.8 or above
[ERROR] Unable to parse source file : 'edelta/src/edelta/util/EdeltaEcoreHelper.java'
[ERROR] Parse error at line 51 column 68: Lambda expressions are allowed only at source level 1.8 or above
[ERROR] Unable to parse source file : 'edelta/src/edelta/interpreter/EdeltaInterpreter.java'
2021-01-13T08:31:39.8779547Z [ERROR] Parse error at line 258 column 11: Multi-catch parameters are not allowed for source level below 1.7
```

--- a/edelta.parent/edelta.maven.plugin/pom.xml
+++ b/edelta.parent/edelta.maven.plugin/pom.xml
@@ -12,9 +12,6 @@
 
   <properties>
     <maven.version>3.6.0</maven.version>
-    <maven.compiler.source>11</maven.compiler.source>
-    <maven.compiler.target>11</maven.compiler.target>
-    <maven.compiler.release>11</maven.compiler.release>
   </properties>
 
   <prerequisites>

--- a/edelta.parent/edelta.maven.plugin/pom.xml
+++ b/edelta.parent/edelta.maven.plugin/pom.xml
@@ -12,6 +12,9 @@
 
   <properties>
     <maven.version>3.6.0</maven.version>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <maven.compiler.release>11</maven.compiler.release>
   </properties>
 
   <prerequisites>

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -334,6 +334,15 @@
     <pluginManagement>
       <plugins>
         <plugin>
+          <groupId>org.eclipse.xtend</groupId>
+          <artifactId>xtend-maven-plugin</artifactId>
+          <configuration>
+            <!-- required since we don't specify maven.compiler.source/target
+              https://github.com/LorenzoBettini/edelta/issues/300 -->
+            <javaSourceVersion>11</javaSourceVersion>
+          </configuration>
+        </plugin>
+        <plugin>
           <!-- We don't use it, since we have edelta-maven-plugin, we leave it here
             just for historical reasons ;) -->
           <groupId>org.eclipse.xtext</groupId>

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -393,8 +393,9 @@
             </execution>
           </executions>
           <configuration>
-            <!-- required since we don't use maven.compiler.source/target
-              https://github.com/LorenzoBettini/edelta/issues/300 -->
+            <!-- That is not required with maven.compiler.source/target set to 11
+              but it's required if we switch to a newer version of Java
+              that is not yet handled by Xbase -->
             <compilerSourceLevel>11</compilerSourceLevel>
             <compilerTargetLevel>11</compilerTargetLevel>
           </configuration>

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -19,6 +19,9 @@
     <assertj-version>3.14.0</assertj-version>
     <mockito-version>3.2.4</mockito-version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
+    <maven.compiler.release>11</maven.compiler.release>
     <jacoco-version>0.8.6</jacoco-version>
     <maven-antrun-plugin.version>1.8</maven-antrun-plugin.version>
 

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -337,8 +337,9 @@
           <groupId>org.eclipse.xtend</groupId>
           <artifactId>xtend-maven-plugin</artifactId>
           <configuration>
-            <!-- required since we don't specify maven.compiler.source/target
-              https://github.com/LorenzoBettini/edelta/issues/300 -->
+            <!-- That is not required with maven.compiler.source/target set to 11
+              but it's required if we switch to a newer version of Java
+              that is not yet handled by Xtend -->
             <javaSourceVersion>11</javaSourceVersion>
           </configuration>
         </plugin>

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -19,9 +19,6 @@
     <assertj-version>3.14.0</assertj-version>
     <mockito-version>3.2.4</mockito-version>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>11</maven.compiler.source>
-    <maven.compiler.target>11</maven.compiler.target>
-    <maven.compiler.release>11</maven.compiler.release>
     <jacoco-version>0.8.6</jacoco-version>
     <maven-antrun-plugin.version>1.8</maven-antrun-plugin.version>
 

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -21,7 +21,6 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
-    <maven.compiler.release>11</maven.compiler.release>
     <jacoco-version>0.8.6</jacoco-version>
     <maven-antrun-plugin.version>1.8</maven-antrun-plugin.version>
 

--- a/edelta.parent/pom.xml
+++ b/edelta.parent/pom.xml
@@ -392,6 +392,12 @@
               </goals>
             </execution>
           </executions>
+          <configuration>
+            <!-- required since we don't use maven.compiler.source/target
+              https://github.com/LorenzoBettini/edelta/issues/300 -->
+            <compilerSourceLevel>11</compilerSourceLevel>
+            <compilerTargetLevel>11</compilerTargetLevel>
+          </configuration>
         </plugin>
         <plugin>
           <groupId>org.jacoco</groupId>

